lib/util/buffered_readable.js

```diff
@@ -31,7 +31,7 @@ BufferedReadable.prototype._read = function() {
   if (this._buffer.length > 0) {
     for (var i = 0; i < this._buffer.length; i++) {
       if (!this.push(this._buffer[i])) {
-        this._buffer = this._buffer.slice(i);
+        this._buffer = this._buffer.slice(i + 1);
```

Comment:
Can you explain why this is correct? Is it because the semantics of the buffer have changed, and it now returns `false` if the push succeeded but no *future* pushes will succeed? Otherwise I fear we are going to remove entries from our internal buffer even if they haven't been successfully pushed into the output stream, and ultimately skip them / drop results.